tools/engine_tool/lib/src/gn_utils.dart

```diff
@@ -106,9 +138,20 @@ Future<Map<String, TestTarget>> findTestTargets(
 /// We support:
 ///   1) Exact label matches (the '//' prefix will be stripped off).
 ///   2) '/...' suffix which selects all targets that match the prefix.
-Set<TestTarget> selectTargets(
-    List<String> selectors, Map<String, TestTarget> allTargets) {
-  final Set<TestTarget> selected = <TestTarget>{};
+///
+/// NOTE: if selectors is empty all targets will be selected.
+Set<BuildTarget> selectTargets(
+    List<String> selectors, Map<String, BuildTarget> allTargets) {
+  final Set<BuildTarget> selected = <BuildTarget>{};
+
+  if (selectors.isEmpty) {
+    // Default to all if no selectors are specified.
+    // TODO(johnmccutchan): Reconsider this default or at least lift
+    // this logic up to the caller.
+    allTargets.values.forEach(selected.add);
+    return selected;
```

Comment:
If you return empty set here, and don't make this an error at the call site of `selectTargets`, then the `extraNinjaTargets` parameter of the `BuildRunner` constructor will be empty, and the targets specified by the build config json will be used.